tests/test_default_values.py

```diff
@@ -0,0 +1,187 @@
+from .data import car
+
+
+class TestDefaultValues(unittest.TestCase):
+    def test_return_empty_containers(self):
+        raw_config = car.build_all_default_config()
+        car_schema = car.build_schema()
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+
+        self.assertTrue(config_suite.valid)
+
+        # Unspecified containers are defaulted to empty container of themselves
+        self.assertEqual(config_suite.snapshot.owner, ())
+        self.assertEqual(config_suite.snapshot.incidents, ())
+
+        self.assertEqual(config_suite.snapshot.color.chassis, None)
+        self.assertEqual(config_suite.snapshot.color.interior, None)
+
+    def test_default_values(self):
+        raw_config = car.build_default_config()
+        car_schema = car.build_schema()
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+        self.assertTrue(config_suite.valid)
+
+        # Named dict top level
+        self.assertEqual(config_suite.snapshot.tire.dimension, 15)
+        self.assertEqual(config_suite.snapshot.tire.rim, "green")
+
+        # Named dict within dict
+        owner_data = (("Johan", "Svalbard"), ("Svein", "Earth"))
+        self.assertEqual(len(owner_data), len(config_suite.snapshot.owner))
+        for (expected_name, expected_location), config_owner in zip(
+            owner_data, sorted(config_suite.snapshot.owner)
+        ):
+            self.assertEqual(expected_name, config_owner.value.name)
+            self.assertEqual(expected_location, config_owner.value.location)
+
+        # Named dict within list
+        incidents_data = (("whereabouts", 0), ("somewhere else", 1))
+        self.assertEqual(len(incidents_data), len(config_suite.snapshot.incidents))
+        for (expected_location, expected_casualties), config_owner in zip(
+            owner_data, sorted(config_suite.snapshot.incidents)
+        ):
+            self.assertEqual(config_owner.location, expected_location)
+            self.assertEqual(config_owner.casualties, expected_casualties)
+
+    def test_transformations_applied(self):
+        raw_config = car.build_all_default_config()
+        car_schema = car.build_schema()
+        car_schema[MK.Content]["tire"][MK.Content]["dimension"][
+            MK.Transformation
+        ] = car._inch_to_cm
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+        self.assertTrue(config_suite.valid)
+        self.assertEqual(config_suite.snapshot.tire.dimension, 17 * 2.54)
+
+    def test_default_require_not_allowed(self):
+        raw_config = car.build_default_config()
+        car_schema = car.build_schema()
+        car_schema[MK.Content]["tire"][MK.Content]["dimension"][MK.Required] = True
+        with self.assertRaises(ValueError) as error_context:
+            configsuite.ConfigSuite(raw_config, car_schema)
+        self.assertTrue(str(error_context.exception).startswith("Is default value"))
+
+    def test_invalid_default_type(self):
```

Comment:
Do we still expect this failure though? Are we doing any default value validation?

Reply:
I'll remove the test and whatever remains of the type validation